configure.ac

```diff
@@ -153,9 +153,17 @@ dnl When bumping the libsoup-2.4 dependency, remember to bump
 dnl SOUP_VERSION_MIN_REQUIRED and SOUP_VERSION_MAX_ALLOWED in
 dnl Makefile.am
 SOUP_DEPENDENCY="libsoup-2.4 >= 2.39.1"
+SOUP3_DEPENDENCY="libsoup-3.0 >= 3.0.0"
 AC_ARG_WITH(soup,
 	    AS_HELP_STRING([--with-soup], [Use libsoup @<:@default=yes@:>@]),
 	    [], [with_soup=$with_soup_default])
+AC_ARG_WITH(soup3,
+	    AS_HELP_STRING([--with-soup3], [Use libsoup3 @<:@default=no@:>@]),
+	    [], [with_soup3=no])
+AS_IF([test x$with_soup3 != xno], [
+    SOUP_DEPENDENCY=$SOUP3_DEPENDENCY
+    with_soup=yes
```

Comment:
I'd prefer if the soup3 handling was distinct from the soup2 handling rather than being an option if possible. In other words, I don't think enabling soup3 should set `with_soup=yes` and I think `OSTREE_FEATURES` should say `libsoup3`.

Reply:
I redid this part so that the soup3 option is distinct just like the curl option.